Source/PlasticSourceControl/Private/SPlasticSourceControlBranchesWidget.cpp

```diff
@@ -553,38 +562,59 @@ TSharedPtr<SWidget> SPlasticSourceControlBranchesWidget::OnOpenContextMenu()
 
 	FToolMenuSection& Section = *Menu->FindSection("Source Control");
 
+	static const FText CreateChildBranchTooltip(LOCTEXT("CreateChildBranchTooltip", "Create child branch from the selected branch."));
+	const FText& CreateChildBranchTooltipDynamic = bSingleSelection ? CreateChildBranchTooltip : SelectASingleBranchTooltip;
 	Section.AddMenuEntry(
 		TEXT("CreateChildBranch"),
 		LOCTEXT("CreateChildBranch", "Create child branch..."),
-		LOCTEXT("CreateChildBranchTooltip", "Create child branch from the selected branch."),
+		CreateChildBranchTooltipDynamic,
 		FSlateIcon(),
 		FUIAction(
 			FExecuteAction::CreateSP(this, &SPlasticSourceControlBranchesWidget::OnCreateBranchClicked, SelectedBranch),
-			FCanExecuteAction::CreateLambda([this, SelectedBranch]() { return (!SelectedBranch.IsEmpty()); })
+			FCanExecuteAction::CreateLambda([bSingleSelection]() { return bSingleSelection; })
 		)
 	);
 
+	static const FText SwitchToBranchTooltip(LOCTEXT("SwitchToBranchTooltip", "Switch the current workspace to this branch."));
+	const FText& SwitchToBranchTooltipDynamic = bSingleNotCurrent ? SwitchToBranchTooltip : bSingleSelection ? SelectADifferentBranchTooltip : SelectASingleBranchTooltip;
 	Section.AddMenuEntry(
 		TEXT("SwitchToBranch"),
 		LOCTEXT("SwitchToBranch", "Switch workspace to this branch"),
-		LOCTEXT("SwitchToBranchTooltip", "Switch workspace to this branch."),
+		SwitchToBranchTooltipDynamic,
 		FSlateIcon(),
 		FUIAction(
 			FExecuteAction::CreateSP(this, &SPlasticSourceControlBranchesWidget::OnSwitchToBranchClicked, SelectedBranch),
-			FCanExecuteAction::CreateLambda([this, SelectedBranch]() { return (!SelectedBranch.IsEmpty()) && (SelectedBranch != CurrentBranchName); })
+			FCanExecuteAction::CreateLambda([bSingleNotCurrent]() { return bSingleNotCurrent; })
+		)
+	);
+
+	Section.AddSeparator("PlasticSeparator1");
+
+	static const FText MergeBranchTooltip(LOCTEXT("MergeBranchTooltip", "Merge this branch into the current workspace."));
+	const FText& MergeBranchTooltipDynamic = bMergeXml ? bSingleNotCurrent ? MergeBranchTooltip : bSingleSelection ? SelectADifferentBranchTooltip : SelectASingleBranchTooltip : UpdateUVCSTooltip;

```

Comment:
Same, could you rewrite this so it's easier to read? 🙏🏻

Reply:
Same, I made a change to have only one test per line, and align them all with same indentation, so this one now reads better in 4 lines

Reply:
Looks good now! https://github.com/PlasticSCM/UEPlasticPlugin/pull/102/commits/eab7076a2aa14b7c7c1db1924b351ecfe4aff854